lntest/itest/lnd_funding_test.go

```diff
@@ -533,16 +529,25 @@ func testExternalFundingChanPoint(ht *lntemp.HarnessTest) {
 	// First, we'll try to close the channel as Carol, the initiator. This
 	// should fail as a frozen channel only allows the responder to
 	// initiate a channel close.
-	ht.CloseChannelAssertErr(carol, chanPoint2, false)
+	err := ht.CloseChannelAssertErr(carol, chanPoint2, false)
+	require.Contains(ht, err.Error(), "cannot co-op close frozen channel")
 
 	// Before Dave closes the channel, he needs to check the invoice is
 	// settled to avoid an error saying cannot close channel due to active
 	// HTLCs.
 	ht.AssertInvoiceSettled(dave, resp.PaymentAddr)
 
+	// TODO(yy): remove the sleep once the following bug is fixed.
+	// When the invoice is reported settled, the commitment dance is not
+	// yet finished, which can cause an error when closing the channel,
+	// saying there's active HTLCs. We need to investigate this issue and
+	// reverse the order to, first finish the commitment dance, then report
+	// the invoice as settled.
+	time.Sleep(2 * time.Second)
```

Comment:
Actually I found this issue after refactoring the `CloseChannel` by removing the `wait.NoError`. Ever since then I suspect the `wait.NoError` is hiding something. Later on, in following PRs, this error popped up multiple times in other tests, which annoyed me a bit, leading me to think maybe I should put it in the `CloseChannelAssertPending`. But I decided to choose not to, and want to mark a TODO whenever I see it fail, mainly due to three reasons,
- other places have a similar error, but a different precondition. In this test it's after the invoice is reported as settled, in other cases, it's after payment is reported as succeeded, or closing channel right after opening. I think they have different causes and want to fix them one by one.
- I want to mark the specific places in the test so it's easier to come back to locate and fix them.
- I intentionally do this ugly and hacky `time.Sleep` style to address what's hidden in a `wait` and more importantly remind myself to fix them asap(the plan is to get itest pass, and start fixing the issues found right after).

Reply:
Hmm, okay. Sounds good to me, as long as we have an easy pattern to search for :+1:

Reply:
Cool will remove this `TODO` once these changes are made.